### ISSUE TITLE
SF connector issues, Failed to resolve endpoint

### ISSUE THREAD

**Issue by tplevko** [syndesisio/syndesis#884]
I created the Sf->DB integration following the steps described in the [fuse ignite sample integration tutorial](https://access.redhat.com/documentation/en-us/red_hat_jboss_fuse/7.0-tp/html-single/fuse_ignite_sample_integration_tutorial/).
The integration was created successfully, but when I try to create a new lead in salesforce, the integration hits following exception:

```
org.apache.camel.component.salesforce.api.SalesforceException: Unhandled exception: Failed to resolve endpoint: salesforce-on-create-component://getSObject?rawPayload=true&sObjectName=Lead due to: Cannot auto create component: salesforce-on-create-component
```

The whole log file is attached.
[test-2.log](https://github.com/syndesisio/syndesis/files/1597681/test-2.log)

**Comment by jimmidyson** [comment]
@tplevko I don't see that first error in the first log, but I do see:

    java.lang.IllegalArgumentException: clientId must be specified

Could you please create the second issue as a separate issue so we can discuss there rather than mixing up the two different issues? Thanks!

**Comment by jimmidyson** [comment]
Could you possibly export the integration and attach it so I can import it and inspect what's been created?

**Comment by tplevko** [comment]
@jimmidyson, I created separate issue for the second problem: https://github.com/syndesisio/syndesis/issues/888
The exported integration:
[sfdbtest-export.zip](https://github.com/syndesisio/syndesis/files/1599779/sfdbtest-export.zip)

**Comment by jimmidyson** [comment]
@dsimansk @tplevko Could you please run through this again to see if it still an issue? Thanks!

**Comment by iocanel** [comment]
@jimmidyson: FWIW, I don't seem to be able to reproduce.

**Comment by iocanel** [comment]
@jimmidyson: Wrong! I just did!

**Comment by jimmidyson** [comment]
Damnit!

**Comment by zregvart** [comment]
Fixed by #975
